### ISSUE TITLE
Input Driven Transitions - learning Ws

### ISSUE THREAD

**Issue by zashwood** [slinderman/ssm#18]
Hi Scott,

I was just wondering if it was intentional that the weights Ws for the InputDrivenTransitions class are not updated during fitting?  In particular, if you modify your first notebook to print the Ws parameters before and after fitting, you can see that they are the same.  This is because the M-Step for InputDrivenTransitions is inherited from the StickyTransitions class.  I suspect this should not be the case, and that Ws should be learned.  Thanks.

`### Set the parameters of the HMM
T = 100    # number of time bins
K = 2       # number of discrete states
D = 1       # data dimension
M = 3 # Input dimension 

### Make an HMM
true_hmm = HMM(K, D, M, observations="categorical",transitions = "inputdriven", observation_kwargs = {"C": 3})

### Sampled input
input = npr.randn(T, M)

### Sample some data from the HMM
z, y = true_hmm.sample(T, input = input)

### True Loglikelihood
true_ll = true_hmm.log_probability(y, inputs = input[1:])

### New HMM
N_iters = 500
hmm = HMM(K, D, M, observations="categorical", transitions = "inputdriven", observation_kwargs = {"C": 3})

### Print Ws before fitting
hmm.params[1][1]

### Fit 
hmm_lls = hmm.fit(y, inputs = input[1:], method="em", num_em_iters=N_iters, verbose=True)

### Print Ws after fitting
hmm.params[1][1]`

**Comment by slinderman** [comment]
Hi Zoe,

Thanks for catching this!  I haven't used the `InputDrivenTransitions` in my own work (only the `RecurrentTransitions`), and this is definitely a bug.  In short, `InputDrivenTransitions.m_step` was automatically inheriting from `StickyTransitions.m_step`, which obviously doesn't update the input weights.  I've fixed this by explicitly calling the `_Transitions.m_step` function, which uses BFGS to optimize the expected log transition probabilities.

While I was at it, I added a notebook example for the input driven HMM. https://github.com/slinderman/ssm/blob/master/notebooks/2%20Input%20Driven%20HMM.ipynb

I also changed the sample function (again) so that it returns a length `T` data array instead of `T-1`.  

The relevant commits are:
https://github.com/slinderman/ssm/commit/c33ee39fed406df8240acc9dadd810a846d71f38
https://github.com/slinderman/ssm/commit/d15f911dd89c55f98f28b2a2adb8e0c79d0f68f1

Hope this works for you.

Scott
